src/hooks/useLoadClientScript.ts

```diff
@@ -9,27 +9,58 @@ export function useLoadClientScript(props: NostoScriptProps) {
   const { host = "connect.nosto.com", account, shopifyMarkets, loadScript = true } = props
   const [clientScriptLoaded, setClientScriptLoaded] = useState(false)
 
-  useEffect(() => {
-    function scriptOnload() {
-      // Override for production scripts to work in unit tests
-      if ("nostoReactTest" in window) {
-        window.nosto?.reload({
-          site: "localhost"
-        })
-      }
-      setClientScriptLoaded(true)
+  function scriptOnload() {
+    // Override for production scripts to work in unit tests
+    if ("nostoReactTest" in window) {
+      window.nosto?.reload({
+        site: "localhost"
+      })
     }
+    setClientScriptLoaded(true)
+  }
 
-    // Create script element
-    function createScriptElement(urlPartial: string) {
-      const scriptEl = document.createElement("script")
-      scriptEl.type = "text/javascript"
-      scriptEl.src = `//${host}${urlPartial}`
-      scriptEl.async = true
-      scriptEl.setAttribute("nosto-client-script", "")
-      return scriptEl
+  // Create and append script element
+  function createScriptElement(urlPartial: string, language?: string, marketId?: string) {
```

Comment:
again a declaration outside of the useEffect block

Reply:
maybe rename function since logic has changed?

Reply:
injectScriptElement would work